src/test/tools/pmemobjcli/shortencommands.py

```diff
@@ -0,0 +1,146 @@
+# modification, are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions of source code must retain the above copyright
+#       notice, this list of conditions and the following disclaimer.
+#
+#     * Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#
+#     * Neither the name of the copyright holder nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import argparse
+import re
+import os.path
+
+POCLI_STRUCTURE_NAME = 'pocli_cmd pocli_commands'
+
+
+def pmem_obj_cli_functions_block(path):
+    """
+    Finds the block containing pocli commands in a file with the given path.
+
+    :param path: Path to the pmemobj_cli.c file.
+    :return: List of lines containing the pocli commands block.
+    """
+    lines = []
+    start_appending = False
+    start_of_command_block = -1
+    end_of_command_block = -1
+    with open(path) as pmem_obj_cli_file:
+        for i, line in enumerate(pmem_obj_cli_file):
+            if POCLI_STRUCTURE_NAME in line:
+                start_of_command_block = i
+                start_appending = True
+            if start_appending:
+                lines.append(line)
+                if '};' in line:
+                    end_of_command_block = i
+            if start_of_command_block > 0 and end_of_command_block > 0:
+                break
+    return lines
+
+
+def parsed_pocli_commands(lines):
+    """
+    Creates a dictionary based on the given block of pocli commands.
+    Long names of the commands are stored as keys and short names are stored as values.
+
+    :param lines: Lines containing the pocli commands block.
+    :type lines: list of str
+    :return: Dictionary with the pocli commands, where long names are stored as keys and their short names as values.
+    """
+    commands = {}
+    for i, line in enumerate(lines):
+        if POCLI_STRUCTURE_NAME not in line and ('{' in line):
```

Comment:
Small explanation:
The block of command is formatted like this:
```
static struct pocli_cmd pocli_commands[] = {
	{
		"help",		/* name */
		"h",		/* name_short */
		"[<cmd>]",	/* usage */
		pocli_help,	/* func */
	},
	{
		"quit",
		"q",
		"",
		pocli_quit,
	},
```
I'm looking for a '{' which represents the start of a single pmemobjcli command and I'm excluding the line with POCLI_STRUCTURE_NAME since it also contains a '{'.

Reply:
if you're commenting on Leszek's comment:
>> "('{' in line)" - these brackets seem to be unnecessary <<
Then explanation is not needed - Leszek is pointing to unnecessary parentheses >>( )<<